libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/PushNotificationManager.swift

```diff
@@ -127,38 +152,99 @@ extension PushNotificationManager {
+              let notificationType = getNotificationType(apiName: notificationTypeApiName, account: account) else {
+            return nil
+        }
+        return notificationType.actionGroups.first { $0.name == actionGroupName }
+    }
+    
+    /// Retrieves a specific action by its identifier within a given notification type.
+    ///
+    /// - Parameters:
+    ///   - notificationTypeApiName: The API name of the notification type.
+    ///   - actionIdentifier: The identifier of the action.
+    ///   - account: The user account from which to retrieve the action. Defaults to the current user.
+    /// - Returns: The `Action` if found, otherwise `nil`.
+    @objc
+    public func getAction(notificationTypeApiName: String,
+                          actionIdentifier: String,
+                          account: UserAccount? = UserAccountManager.shared.currentUserAccount) -> Action? {
+        guard let account = account,
+              let notificationType = getNotificationType(apiName: notificationTypeApiName, account: account) else {
+            return nil
+        }
+        return notificationType.actionGroups
+            .flatMap { $0.actions }
+            .first { $0.identifier == actionIdentifier }
+    }
+    
+    /// Invokes a server-side notification action for a specific notification.
     ///
-    /// If the API fetch fails, the function attempts to retrieve notification types from the cache.
-    /// If neither source provides data, the function returns `false` with an appropriate error.
+    /// - Parameters:
+    ///   - client: The `RestClient` instance to use for the request. Defaults to `RestClient.shared`.
+    ///   - notificationId: The ID of the notification for which the action is invoked.
+    ///   - actionIdentifier: The identifier of the action to be performed.
+    /// - Throws: `PushNotificationManagerError.notificationActionInvocationFailed` if the action invocation fails.
+    /// - Returns: An `ActionResultRepresentation` containing the response from the server.
     @objc
-    public func getNotificationTypes(restClient: RestClient = RestClient.shared,
-                                     account: UserAccount) async -> (Bool, Error?) {
+    public func invokeServerNotificationAction(client: RestClient = RestClient.shared,
+                                               notificationId: String,
+                                               actionIdentifier: String
+    ) async throws -> ActionResultRepresentation {
+        let path = "/connect/notifications/\(notificationId)/actions/\(actionIdentifier)"
+        let request = RestRequest(method: .POST, path: path, queryParams: nil)
+        
         do {
-            let types = try await fetchNotificationTypesFromAPI(with: restClient)
-            storeNotification(types: types, with: account)
-            setNotificationCategories(types: types)
-            return (true, nil)
+            let response = try await client.send(request: request)
```

Comment:
Check that api version is 64 or more (see https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/v12.2.0/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFPushNotificationManager.m#L157).